src/etc/_cargo

```diff
@@ -1,482 +1 @@
-    flags=( help ${${${(M)${(f)"$(_call_program flags cargo -Z help)"}:#*--*}/ #-- #/:}##*-Z } )
-    _describe -t flags 'unstable flag' flags
-}
-
-_cargo_installed_crates() {
-    local expl
-    _description crates expl 'crate'
-    compadd "$@" "$expl[@]" - ${${${(f)"$(cargo install --list)"}:# *}%% *}
-}
-
-_cargo_cmds() {
-    local -a commands
-    # This uses Parameter Expansion Flags, which are a built-in Zsh feature.
-    # See more: http://zsh.sourceforge.net/Doc/Release/Expansion.html#Parameter-Expansion-Flags
-    # and       http://zsh.sourceforge.net/Doc/Release/Expansion.html#Parameter-Expansion
-    #
-    # # How this work?
-    #
-    # First it splits the result of `cargo --list` at newline, then it removes the first line.
-    # Then it removes indentation (4 whitespaces) before each items. (Note the x## pattern [1]).
-    # Then it replaces those spaces between item and description with a `:`
-    #
-    # [1]: https://github.com/zsh-users/zsh-completions/blob/master/zsh-completions-howto.org#patterns
-    commands=( ${${${(M)"${(f)$(_call_program commands cargo --list)}":#    *}/ ##/}/ ##/:} )
-    _describe -t commands 'command' commands
-}
-
-_cargo_target_triple() {
-    local -a result
-
-    if (( $+commands[rustup] )); then
-        result=( ${(f)"$(rustup target list --installed)"} )
-    else
-        result=( ${(f)"$(rustc --print target-list)"} )
-    fi
-
-    _describe 'target triple' result
-}
-
-#FIXME: Disabled until fixed
-#gets package names from the manifest file
-_cargo_package_names() {
-    _message -e packages package
-}
-
-# Extracts the values of "name" from the array given in $1 and shows them as
-# command line options for completion
-_cargo_names_from_array() {
-    local manifest=$(cargo locate-project --message-format plain)
-    if [[ -z $manifest ]]; then
-        return 0
-    fi
-
-    local last_line
-    local -a names;
-    local in_block=false
-    local block_name=$1
-    names=()
-    while read -r line; do
-        if [[ $last_line == "[[$block_name]]" ]]; then
-            in_block=true
-        else
-            if [[ $last_line =~ '\s*\[\[.*' ]]; then
-                in_block=false
-            fi
-        fi
-
-        if [[ $in_block == true ]]; then
-            if [[ $line =~ '\s*name\s*=' ]]; then
-                regexp-replace line '^\s*name\s*=\s*|"' ''
-                names+=( "$line" )
-            fi
-        fi
-
-        last_line=$line
-    done < "$manifest"
-    _describe "$block_name" names
-
-}
-
-#Gets the test names from the manifest file
-_cargo_test_names() {
-    _cargo_names_from_array "test"
-}
-
-#Gets the bench names from the manifest file
-_cargo_benchmark_names() {
-    _cargo_names_from_array "bench"
-}
-
-_cargo_example_names() {
-    if [[ -d examples ]]; then
-        local -a files=(${(@f)$(echo examples/*.rs(:t:r))})
-        _values 'example' "${files[@]}"
-    fi
-}
-
-_cargo
+echo "source <(cargo complete --shell zsh)" >> ~/.zshrc
```

Comment:
```suggestion
source <(cargo complete --shell zsh)
```

If someone is using this file, it will already be registered

Reply:
Thanks!